src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs

```diff
@@ -307,7 +308,23 @@ public Task<Compilation> GetCompilationAsync(Solution solution, CancellationToke
                 }
                 else
                 {
-                    return GetOrBuildCompilationAsync(solution, lockGate: true, cancellationToken: cancellationToken);
+                    return GetOrBuildCompilationInfoAsync(solution, lockGate: true, cancellationToken: cancellationToken)
+                        .ContinueWith(t => t.Result.Compilation, cancellationToken, TaskContinuationOptions.ExecuteSynchronously | TaskContinuationOptions.OnlyOnRanToCompletion, TaskScheduler.Default);
+                }
+            }
+
+            public Task<bool> GetCompilationHasIncompleteReferencesAsync(Solution solution, CancellationToken cancellationToken)
+            {
+                var state = this.ReadState();
+
+                if (state.HasIncompleteReferences.HasValue)
```

Comment:
This is *very* weird.  Say we've only computed up to the 'in progress' state.  From your code elsewhere, you initialize this state with "true" for "HasIncompleteReferences".

But this doesn't seem accurate to me.  You're effectively saying "the compilation is missing references!".  But if I actually went and then asked for the compilation it might have all references.  

Furthermore, it now seems like the Project might return different results for this method depending on when it was called.  i.e. if some other thread took us to the "final state" or not.  That makes this API seem observably mutable, and I'm really wary about having that in the API.